File: java/BuilderBolo.java
public class BuilderBolo implements Builder {

    private Bolo bolo;

    public BuilderBolo() {
        this.bolo = new Bolo();
    }

    @Override
    public Builder adicionarCobertura(String cobertura) {
        bolo.setCobertura(cobertura);
        return this;
    }

    @Override
    public Builder adicionarRecheio(String recheio) {
        bolo.setRecheio(recheio);
        return this;
    }

    @Override
    public Builder adicionarTamanho(String tamanho) {
        bolo.setTamanho(tamanho);
        return this;
    }

    @Override
    public Bolo getBolo() {
        return this.bolo;
    }

}

File: php/BuilderBolo.php
<?php

class BuilderBolo implements Builder
{

    private $bolo;

    public function __construct()
    {
        $this->reset();
    }

    public function reset(): void
    {
        $this->bolo = new Bolo();
    }
    public function adicionarCobertura($cobertura): void
    {
        $this->bolo->cobertura = $cobertura;
    }

    public function adicionarRecheio($recheio): void
    {
        $this->bolo->recheio = $recheio;
    }

    public function adicionarTamanho($tamanho): void
    {
        $this->bolo->tamanho = $tamanho;
    }

    public function getBolo(): Bolo
    {
        $resultado = $this->bolo;
        $this->reset();
        return $resultado;
    }
}
